services/wallet/token/token.go

```diff
@@ -289,6 +290,22 @@ func (tm *Manager) FindOrCreateTokenByAddress(ctx context.Context, chainID uint6
 	return token
 }
 
+func (tm *Manager) MarkAsPreviouslyOwnedToken(token *Token, owner common.Address) error {
+	if token == nil {
+		return errors.New("token is nil")
+	}
+	if (owner == common.Address{}) {
+		return errors.New("owner is nil")
+	}
+	count := 0
+	err := tm.db.QueryRow(`SELECT count(*) FROM token_balances WHERE user_address = ? AND token_address = ? AND chain_id = ?`, owner.Hex(), token.Address.Hex(), token.ChainID).Scan(&count)
```

Comment:
premature optimization incoming: if you only need to know if there's more than 0 rows, there's a more efficient query:
https://stackoverflow.com/questions/9755860/valid-query-to-check-if-row-exists-in-sqlite3